lib/Parse/ParseDecl.cpp

```diff
@@ -2906,27 +2906,43 @@ ParserStatus Parser::parseNewDeclAttribute(DeclAttributes &Attributes,
     }
     
     // Parse the subject.
-    if (Tok.isContextualKeyword("set")) {
-      consumeToken();
-    } else {
-      diagnose(Loc, diag::attr_access_expected_set, AttrName);
+    if (!Tok.isContextualKeyword("set")) {
+      auto diag = diagnose(Loc, diag::attr_access_expected_set, AttrName);
+      
+      if (Tok.is(tok::l_paren)) {
+        return makeParserSuccess();
```

Comment:
Special handling `private((`. Could you explain what this is for?

Reply:
The previous code already checked for the current token not being `(` before consuming a `)` as it wouldn't be closing the `(` of the access modifier subject as expected in the recovery.

I thought I'd add the same check in the case of `private(( <declaration>` but thinking about it again it might not be necessary and we can still just suggest to replace the extra `(` with `set)` (we can also do the same for `private(()`, no?).

What do you think?

Reply:
It seems like the previous code was assuming if you had `private( ()` then the `()` was something unrelated to the `private(`, i.e the closing `)` matches the `(` in the `()`, not the `private(`. That seems somewhat reasonable to me, although I doubt this case happens often in practice. I think we should probably move this check to the `if (peekToken().is(tok::r_paren))` condition, that way we can still correct `private(( <declaration>`.